### ISSUE TITLE
ERROR    HTTPStatus Exception: Client error '429 Too Many Requests' for url 'https://api.openai.com/v1/chat/completions' For more information check: https://httpstatuses.com/429

### ISSUE THREAD

**Issue by ataur39n-sharif** [eli64s/readme-ai#61]
I am using this with Docker. Here I providing some error messages--

Error - 

ERROR    HTTPStatus Exception:
Client error '429 Too Many Requests' for url 'https://api.openai.com/v1/chat/completions'
For more information check: https://httpstatuses.com/429

images -

![doc-command](https://github.com/eli64s/readme-ai/assets/76764955/106ec9cd-36db-4dbd-ab55-25bc4a53e2a7)
![doc-0](https://github.com/eli64s/readme-ai/assets/76764955/983f03b0-c239-4dea-99d8-3a844c745b52)
![doc-1](https://github.com/eli64s/readme-ai/assets/76764955/72d861be-ce0c-4aa2-8204-7a39e4eb13c0)
![doc-2](https://github.com/eli64s/readme-ai/assets/76764955/96c4ee8d-b785-4d8f-9171-ab6facc78f02)

![doc-command](https://github.com/eli64s/readme-ai/assets/76764955/106ec9cd-36db-4dbd-ab55-25bc4a53e2a7)

**Comment by eli64s** [comment]
Hi @ataur39n-sharif I just pulled the latest image and ran you're repo. Can you try once more with the latest and let me know if you still experience this?

Thanks!

**Comment by Cro22** [comment]
Same Here! I tried with the latest image.
![image](https://github.com/eli64s/readme-ai/assets/47543878/c1ea4d02-ca05-4544-98c9-10d99c1e48b7)

**Comment by eli64s** [comment]
@Cro22 @ataur39n-sharif Are you using a free OpenAI account or payment method?

**Comment by Cro22** [comment]
@eli64s I use the paid API

**Comment by ataur39n-sharif** [comment]
@eli64s I am using a free account of OpenAI

**Comment by jatolentino** [comment]
I'm also getting the same 429 error on my readme using [https://readmeai.streamlit.app/]

**Comment by Aviksaikat** [comment]
why not add rate limiting ?

**Comment by eli64s** [comment]
@Aviksaikat There is a default rate limit setting in the [config file](https://github.com/eli64s/readme-ai/blob/main/readmeai/settings/config.toml#L6)

This seems like a common issue for unpaid accounts, but still happens for paid accounts occasionally. I may need to work on a more robust API implementation to solve this problem for everyone.

**Comment by jatolentino** [comment]
@eli64s should we increase or decrease the rate_limit variable?

**Comment by Aviksaikat** [comment]
it should be decreased.

**Comment by Aviksaikat** [comment]
how can we update the config file ?

**Comment by Aviksaikat** [comment]
that rate_limit field it totally useless.

**Comment by daan-ef2** [comment]
Got the same error and using a paid api-key. Is there any workaround?

**Comment by eli64s** [comment]
@abhi245y that is correct, using the model engine `gpt-4-1106-preview` would be a temporary workaround. However, take note of the following before trying this.

> [!WARNING]
>
> During brief testing of the `gpt-4-1106-preview` model I've noticed higher API costs.
> If trying this workaround, use the [OpenAI API Dashboard](https://platform.openai.com/usage) to continuously track your API usage and cost.

Thank you,
*Eli*

**Comment by abhi245y** [comment]
Yah you are right, when I gave the script a few runs during testing, my usage shot up from 0.21 to 1.4 real quick. But if you're just using it once, it's no big deal.

I also noticed that you switched the model to gpt-4-1106-preview and then reverted it back to gpt-4. At first, I didn't understand why, but now it makes sense.

**Comment by alexiuscrow** [comment]
Had issues with `—model gpt-4-1106-preview` and `—model gpt-3.5-turbo`, but it works well with `—model gpt-4`.

**Comment by Just-Goo** [comment]
> @abhi245y that is correct, using the model engine `gpt-4-1106-preview` would be a temporary workaround. However, take note of the following before trying this.
> 
> Warning
> 
> During brief testing of the `gpt-4-1106-preview` model I've noticed higher API costs. If trying this workaround, use the [OpenAI API Dashboard](https://platform.openai.com/usage) to continuously track your API usage and cost.
> 
> Thank you, _Eli_

@eli64s  hi I'm using docker to run 'readme-ai'. Please what's the workaround I can use for this error

**Comment by warmfire540** [comment]
rate limit code seems to just be a semephore, not RPM like we are wanting..
